webapp/channels/src/utils/plugins/plugin_setting_extraction.test.tsx

```diff
@@ -81,15 +83,80 @@ function getFullExample(): PluginConfiguration {
         icon: 'some icon',
     };
 }
+
+function CustomSection() {
+    return (
+        <div>{'Custom Section'}</div>
+    );
+}
+
+function CustomInput() {
+    return (
+        <input>{'Custom Input'}</input>
+    );
+}
+
+function getCustomExample(): PluginConfiguration {
+    return {
+        id: '',
+        uiName: 'some name',
+        sections: [
+            {
+                title: 'Section',
+                settings: [
+                    {
+                        name: 'radioA',
+                        options: [
+                            {
+                                text: 'Enabled',
+                                value: 'on',
+                            },
+                            {
+                                text: 'Disabled',
+                                value: 'off',
+                            },
+                        ],
+                        type: 'radio',
+                        default: 'off',
+                    },
+                ],
+            },
+            {
+                title: 'Section with custom setting',
+                settings: [
+                    {
+                        name: 'custom_input',
+                        type: 'custom',
+                        component: CustomInput,
+                    },
+                ],
+            },
+            {
+                title: 'Custom section',
+                component: CustomSection,
+            },
+        ],
+    };
+}
+
 describe('plugin setting extraction', () => {
+    beforeAll(() => {
+        console.warn = jest.fn();
```

Comment:
@larkox To be honest I added that to avoid getting tests to fail because of: 

https://github.com/mattermost/mattermost/blob/b596430920a3d95ce270b14497de6e6ef918ab46/webapp/channels/src/tests/setup_jest.ts#L96-L101

Ideally, we should assert whether `console.warn` gets called in case we filter out anything due to a failure during validation/extraction.

Reply:
Regardless, we should probably reset it once done. I'll send an update.

Reply:
Those lines in `setup_jest.ts` frequently cause me problems. I'm really thinking of getting rid of that stuff 😛